### PR TITLE
Add fetching of userList GQL endpoint

### PR DESCRIPTION
**Pull request by LucoLe**
#### Summary

Adds fetchPublicUserListsById GQL endpoint.

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

#### Screenshots

image

### CODE REVIEW

File: lib/sanbase_web/graphql/resolvers/user_list_resolver.ex [hunk 1]
```diff
@@ -104,6 +104,19 @@ defmodule SanbaseWeb.Graphql.Resolvers.UserListResolver do
     end
   end
 
+  def fetch_public_user_lists_by_id(_root, %{user_list_id: user_list_id}, _resolution) do
+    case UserList.fetch_public_user_lists_by_id(user_list_id) do
+      {:ok, user_list} ->
+        {:ok, user_list}
+
+      {:error, changeset} ->
```
Comment (tspenov) [inline lib/sanbase_web/graphql/resolvers/user_list_resolver.ex]: I think this is not needed cause it returns `{:ok, nil}` || `{:ok, user_list}`.
Also we should probably mention this in the description so @frontend knows how it behaves

**Comment by tspenov** [comment]
Other than the upper comment - LGTM

**Comment by IvanIvanoff** [comment]
@LucoLe here you get a single list by id so it makes sense to name the query with a singular `list` instead of `lists`. Also it's understandable by the params that it's by id so `fetchUserList` or even `userList` should be ok.
Also the `public` in the name is misleading - a user should be able to fetch his or her own private list by id.

**Comment by LucoLe** [comment]
I'd like not to mix the public lists with the private ones. IMHO if the user needs to fetch his private list by id let's add another endpoint in another PR.

**Comment by IvanIvanoff** [comment]
I do not fully agree with you. It will make the API interaction much more complicated. A simple check as:
`where user_list_public = true OR (user_list_public = false and user_list_owner = user_id)` will cover all cases

**Comment by LucoLe** [comment]
I'm not saying it's tough to achieve I just think it will make the api not that clean but if you insist I'll put it there.

**Comment by IvanIvanoff** [comment]
Let's see the opinion of @Partysun and @valo on the topic

File: lib/sanbase_web/graphql/resolvers/user_list_resolver.ex [hunk 2]
```diff
@@ -104,6 +104,12 @@ defmodule SanbaseWeb.Graphql.Resolvers.UserListResolver do
     end
   end
 
+  def user_list(_root, %{user_list_id: user_list_id}, %{
+        context: %{auth: %{current_user: current_user}}
```
[the remark this answers is not in the document]
Comment (Partysun) [reply lib/sanbase_web/graphql/resolvers/user_list_resolver.ex]: Anonymous users have to see public watch lists.

**Comment by Partysun** [comment]
I am not sure, why do we miss watch or assets in the name of the query.
Because a customer of this API request will fetch user's public watch list by id. Userlist sounds like list of users. But this request gives smth different.

In my opinion better to have: fetchWatchList or watchlist or assetslist.
Or maybe I make a mistake, I am looking from mobile GitHub.

**Comment by LucoLe** [comment]
@Partysun That's a valid comment but this is a bigger problem and needs more refactoring because we have a lot of other references to `user_list`. Let's agree to create a card for refactoring all references for `user_list` to `watchlist` and keep the name in this PR. Do you agree @Partysun ?

**Comment by Partysun** [comment]
Agree

**Pull request merged**